app/services/curation_concerns/workflow/permission_query.rb

```diff
@@ -257,7 +257,10 @@ def scope_users_for_entity_and_roles(entity:, roles:)
             .and(entity_responsibilities[:entity_id].eq(entity.id))
         )
 
-        sub_query_for_user = agent_table.project(agent_table[:proxy_for_id]).where(
+        # PostgreSQL requires an explicit cast from string to integer
+        cast = Arel::Nodes::NamedFunction.new "CAST", [agent_table[:proxy_for_id].as("integer")]
```

Comment:
Ah, sorry, I didn't see that this was simply backporting a bug fix, and we certainly should just ship it. I don't know where we  ought to move this conversation, though.